pymc_marketing/clv/distributions.py

```diff
@@ -741,3 +742,148 @@ def logp(value, a, b, r, alpha, T):
             r > 0,
             msg="a > 0, b > 0, alpha > 0, r > 0",
         )
+
+
+class ModifiedBetaGeoNBDRV(RandomVariable):
+    name = "mbg_nbd"
+    signature = "(),(),(),(),()->(2)"
+
+    dtype = "floatX"
+    _print_name = ("ModifiedBetaGeoNBD", "\\operatorname{ModifiedBetaGeoNBD}")
+
+    def __call__(self, a, b, r, alpha, T, size=None, **kwargs):
+        return super().__call__(a, b, r, alpha, T, size=size, **kwargs)
+
+    @classmethod
+    def rng_fn(cls, rng, a, b, r, alpha, T, size):
+        if size is None:
+            size = np.broadcast_shapes(a.shape, b.shape, r.shape, alpha.shape, T.shape)
+
+        a = np.asarray(a)
+        b = np.asarray(b)
+        r = np.asarray(r)
+        alpha = np.asarray(alpha)
+        T = np.asarray(T)
+
+        if size == ():
+            size = np.broadcast_shapes(a.shape, b.shape, r.shape, alpha.shape, T.shape)
+
+        a = np.broadcast_to(a, size)
+        b = np.broadcast_to(b, size)
+        r = np.broadcast_to(r, size)
+        alpha = np.broadcast_to(alpha, size)
+        T = np.broadcast_to(T, size)
+
+        output = np.zeros(shape=size + (2,))  # noqa:RUF005
+
+        lam = rng.gamma(shape=r, scale=1 / alpha, size=size)
+        p = rng.beta(a=a, b=b, size=size)
+
+        def sim_data(lam, p, T):
+            t = 0  # recency
+            n = 0  # frequency
+
+            churn = p  # MBG/NBD customer active with probability p at time 0
+            wait = rng.exponential(scale=1 / lam)
+
+            while t + wait < T and not churn:
+                churn = rng.random() < p
+                n += 1
+                t += wait
+                wait = rng.exponential(scale=1 / lam)
+
+            return np.array([t, n])
+
+        for index in np.ndindex(*size):
+            output[index] = sim_data(lam[index], p[index], T[index])
+
+        return output
```

Comment:
Will need to modify this to reflect the modeling assumptions of `ModifiedBetaGeoModel`. The only difference between this and the one you wrote for `BetaGeoModel` is the starting value of `churn`.